src/test/groovy/com/gooddata/executeafm/afm/SimpleMeasureDefinitionTest.groovy

```diff
@@ -50,15 +50,19 @@ class SimpleMeasureDefinitionTest extends Specification {
         SimpleMeasureDefinition definition = readObjectFromResource("/$SIMPLE_MEASURE_DEFINITION_FULL_JSON", SimpleMeasureDefinition)
 
         then:
-        with(definition.item as UriObjQualifier) {
+        with(definition?.item as UriObjQualifier) {
             uri == QUALIFIER.uri
         }
-        definition.aggregation == 'avg'
-        definition.computeRatio
-        with(definition.filters[0] as PositiveAttributeFilter) {
-            (it.displayForm as UriObjQualifier).uri == QUALIFIER.uri
+        definition?.aggregation == 'avg'
+        definition?.computeRatio
+        with(definition?.filters?.find { it.class == PositiveAttributeFilter }) {
+            it.displayForm?.uri == QUALIFIER.uri
             it.in == ['foo']
         }
+        with(definition?.filters?.find { it.class == NegativeAttributeFilter }) {
```

Comment:
not sure about `.?` here - such test won't fail when the definition or filters are null - do you want that?

Reply:
so what's better:
* `NullPointerException` or
* `Target of 'with' block must not be null`